e2e/crm4/provider/submit-an-application.spec.js

```diff
@@ -0,0 +1,164 @@
+			const clientDetailsPage = new ClientDetailsPage(page);
+			// Actions
+			await clientDetailsPage.fillClientDetailsForm();
+			// Expectations
+			await expect(page.getByRole('heading', { name: 'Your application progress' })).toBeVisible();
+			// Set LAA reference for future use
+			const asideLocator = await page.locator('.aside-task-list');
+			const asideText = await asideLocator.textContent();
+			const laaReference = asideText.split('LAA reference')[1].trim().split('\n')[0];
+			await saveToStorage('laaReference', laaReference);
+			// Actions
+			await page.getByRole('link', { name: 'Case and hearing details' }).click();
+		});
+
+		await test.step('Fill in the case and hearing details', async () => {
+			const caseAndHearingDetailsPage = new CaseAndHearingDetailsPage(page);
+			// Actions
+			await caseAndHearingDetailsPage.fillCaseAndHearingDetailsForm('No');
+			// Expectations
+			await expect(page.getByRole('heading', { name: 'Your application progress' })).toBeVisible();
+			// Actions
+			await page.getByRole('link', { name: 'Primary quote' }).click();
+		});
+
+		await test.step('Fill in the primary quote', async () => {
+			const primaryQuotePage = new PrimaryQuotePage(page);
+			// Actions
+			await primaryQuotePage.fillPrimaryQuoteForm();
+			// Expectations
+			await expect(page.getByRole('heading', { name: 'Service cost' })).toBeVisible();
+		});
+
+		await test.step('Fill in the service cost', async () => {
+			const serviceCostPage = new ServiceCostPage(page);
+			// Actions
+			await serviceCostPage.fillServiceCostForm();
+			// Expectations
+			await expect(page.getByRole('heading', { name: 'Primary quote summary' })).toBeVisible();
+			// Actions
+			await page.getByRole('link', { name: 'Save and continue' }).click();
+			// Expectations
+			await expect(page.getByRole('heading', { name: 'Your application progress' })).toBeVisible();
+			// Actions
+			await page.getByRole('link', { name: 'Alternative quotes' }).click();
+		});
+
+		await test.step('Fill in the alternative quotes', async () => {
+			const alternativeQuotesPage = new AlternativeQuotesPage(page);
+			// Actions
+			await alternativeQuotesPage.fillAlternativeQuotesForm();
+			// Expectations
+			await expect(page.getByRole('heading', { name: 'Your application progress' })).toBeVisible();
+			// Actions
+			await page.getByRole('link', { name: 'Reason for prior authority' }).click();
+		});
+
+		await test.step('Fill in the reason for prior authority', async () => {
+			const reasonWhyPage = new ReasonWhyPage(page);
+			// Actions
+			await reasonWhyPage.fillReasonWhyForm();
+			// Expectations
+			await expect(page.getByRole('heading', { name: 'Your application progress' })).toBeVisible();
+			// Actions
+			await page.getByRole('link', { name: 'Submit application' }).click();
+			// Expectations
+			await expect(page.getByRole('heading', { name: 'Check your answers' })).toBeVisible();
+		});
+
+		await test.step('Check your answers', async () => {
```

Comment:
Maybe this test could be called 'Check your answers and submit'?

Reply:
Creating another PR for restructuring. will rename there.